extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/BarrageUtil.java

```diff
@@ -192,13 +194,33 @@ public static Stream<Field> columnDefinitionsToFields(
             @Nullable final MutableInputTable inputTable,
             @NotNull final Collection<ColumnDefinition<?>> columnDefinitions,
             @NotNull final Function<String, Map<String, String>> fieldMetadataFactory,
+            @NotNull final Map<String, Object> attributes,
             final boolean columnsAsList) {
         // Find the format columns
         final Set<String> formatColumns = new HashSet<>();
         columnDefinitions.stream().map(ColumnDefinition::getName)
                 .filter(ColumnFormatting::isFormattingColumn)
                 .forEach(formatColumns::add);
 
+        // Find columns that are sortable
+        List<String> sortableColumns = new ArrayList<>();
+        List<ColumnDefinition<?>> columnsToCheck = new ArrayList<>(columnDefinitions);
+
+        // limits to restrictedSortTo columns if applicable
+        if (attributes.containsKey(GridAttributes.SORTABLE_COLUMNS_ATTRIBUTE)) {
+            String[] restrictedSortColumns =
+                    attributes.get(GridAttributes.SORTABLE_COLUMNS_ATTRIBUTE).toString().split(",");
+            columnsToCheck.removeIf(
+                    col -> Arrays.stream(restrictedSortColumns).noneMatch(name -> name.equals(col.getName())));
+        }
+
+        columnsToCheck.forEach(col -> {
+            Class<?> dataType = col.getDataType();
+            if (dataType.isPrimitive() || Comparable.class.isAssignableFrom(dataType)) {
+                sortableColumns.add(col.getName());
```

Comment:
We shouldn't be adding columns if the sort is restricted, but rather further restricting them.

I'm thinking something like this:
```java
        Set<String> sortableColumns;
        if (attributes.containsKey(GridAttributes.SORTABLE_COLUMNS_ATTRIBUTE)) {
            final String[] restrictedColumns = attributes.get(GridAttributes.SORTABLE_COLUMNS_ATTRIBUTE).toString().split(",");
            sortableColumns = Arrays.stream(restrictedColumns)
                    .filter(columnName -> isDataTypeSortable(tableDefinition.getColumn(columnName).getDataType()))
                    .collect(Collectors.toSet());
        } else {
            sortableColumns = tableDefinition.getColumns().stream()
                    .filter(columnDefinition -> isDataTypeSortable(columnDefinition.getDataType()))
                    .map(ColumnDefinition::getName)
                    .collect(Collectors.toSet());
        }
```

Where:
```java
    private static boolean isDataTypeSortable(final Class<?> dataType) {
        return dataType.isPrimitive() || Comparable.class.isAssignableFrom(dataType);
    }
```